app/models/assignment.rb

```diff
@@ -79,6 +79,7 @@ class Assignment < Assessment
   has_many :starter_file_groups, dependent: :destroy, inverse_of: :assignment, foreign_key: :assessment_id
 
   after_create :create_autotest_dirs
+  after_create :create_autotest_test_student, if: -> { self.type == 'Assignment' }
```

Comment:
only really needed if tests are enabled for this assignment. Think about a better place to create the relevant objects (maybe in a controller method).